docs/fundamentals/executor/instrumenting-executor.md

```diff
@@ -0,0 +1,267 @@
+
+Every method that is decorated using the {class}`~jina.requests` decorator adds a 
+default tracing span for the defined operation. In addition the operation span context 
+is propagated to the method for creating further user defined child spans within the 
+method.
+
+Custom spans can be created to observe the operation individual steps or record various details and attributes with a finer granularity. When tracing is enabled, Jina provides the OpenTelemetry Tracer implementation as an Executor class attribute that can be used to create new child spans. The `tracing_context` method argument contains the parent span context using which a new span can be created to trace the desired operation in the method.
+
+If tracing is enabled, each executor exports its traces to the configured exporter host via the [Span Exporter](https://opentelemetry.io/docs/reference/specification/trace/sdk/#span-exporter) which are combined by the backend for visualization and alerting purposes.
+
+
+### Create custom traces
+
+A `request` method is the public method used to expose the operation as an API. Depending on the complexity, the method can often be composed of different sub operations that is required to build the final response. 
+
+Each internal step along with its global or request specific attributes can be recorded/observed giving a more finer grained view of the operation at the request level. This helps to quickly identify bottlenecks and isolate request patterns that are causing service degradation or errors.
+
+The `self.tracer` class attribute can be used to create a new child span using the `tracing_context` method argument as follows:
+
+```python
+from jina import Executor, requests
+
+
+class MyExecutor(Executor):
+    @requests
+    def foo(self, docs, tracing_context, **kwargs):
+        with self.tracer.start_as_current_span(
+            'process_docs', context=tracing_context
+        ) as process_span:
+            process_span.set_attribute('sampling_rate', 0.01)
+            docs = process(docs)
+            with self.tracer.start_as_current_span('update_docs') as update_span:
+                try:
+                    update_span.set_attribute('len_updated_docs', len(docs))
+                    docs = update(docs)
+                except Exception as ex:
+                    update_span.set_status(Status(StatusCode.ERROR))
+                    update_span.record_exception(ex)
+```
+
+The above pieces of instrumentation generate 3 spans:
+ 1. Default span with name `foo` for the overall method.
+ 2. `process_span` that measures the `process` and `update` sub operations along with a `sampling_rate` attribute that might be a constant or specific to the request/operation.
+ 3. `update_span` that measures the `updated` operation along with any exception that might arise during the operation. The exception is recorded and marked on the `update_span` span. Since the exception is swallowed, the request succeeds with successful parent spans.
```

Comment:
- It should be "successful".
- Yes and No. The `Exception` class supports multiple exceptions which is caught by the `except` block. During the execution there can be at most one top level exception that is provided by the `ex` variable. Plural can be used for the first sentence and the singular needs to be used for the rest of the sentences because the the code captures the exception occurring at that point of time.